src/main/java/bruh/parser/Parser.java

```diff
@@ -0,0 +1,88 @@
+package bruh.parser;
+
+import bruh.command.AddTaskCommand;
+import bruh.command.Command;
+import bruh.command.DeleteTaskCommand;
+import bruh.command.ExitCommand;
+import bruh.command.ListTasksByDateCommand;
+import bruh.command.ListTasksBySearchCommand;
+import bruh.command.ListTasksCommand;
+import bruh.command.MarkTaskDoneCommand;
+import bruh.exception.BruhException;
+import bruh.exception.InvalidArgumentException;
+import bruh.exception.MissingArgumentException;
+import bruh.task.Task;
+
+/**
+ * Parses strings into commands & performs validation of
+ * the arguments provided.
+ */
+public class Parser {
+    private static final String STOP_SIGNAL = "bye";
+
+    /**
+     * Parses string input and returns the appropriate command
+     * to be run by the chatbot.
+     *
+     * @param input The input entered by the user.
+     * @return The command to be run by the chatbot.
+     */
+    public static Command parseInputToCommand(String input) throws BruhException {
+        if (input.isEmpty()) {
+            throw new InvalidArgumentException();
+        } else if (input.equals(STOP_SIGNAL)) {
+            return new ExitCommand();
+        } else if (input.equals("list")) {
+            return new ListTasksCommand();
+        }
```

Comment:
Perhaps you can consider adding the `else` branch to detect errors as mentioned in the code quality section of the textbook [here](https://nus-cs2103-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#use-the-default-branch).

Reply:
Thanks for raising this up! My rationale for not adding an else branch here is that I want to allow the control to simply proceed to the subsequent `switch` statements, where I included a `default` case. I believe this will catch any inputs that do not match any of the cases & simply throw an `InvalidArgumentException`.